Update Haskell package set to LTS 16.6 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2020-07-24 20:00 +02:00. You can watch this live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR that viewers can use to chat with me and with each other.

Having said all that, if anyone wants to merge this PR sooner (@nh2, @cdepillabout, @maralorn, ...), please feel free to do so! There's no need to wait for me if you don't want to.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-10-gf433b07 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/cae4f643586655ac8db13ee14f728a3373976b7b.

peti committed:
LTS Haskell 16.6

414owen committed:
haskell.packages.ghc865.rebase: jailbreak on ghc 8.6

utdemir committed:
nix-tree: Expose as a top-level attribute, add maintainer

zhenyavinogradov committed:
haskellPackages.xrefcheck: unmark as broken

maralorn committed:
haskellPackages.hnix: Fix build

Add override as explained in the comment

maralorn committed:
ghc.withPackages: Wrap hls with package lookup env

maralorn committed:
haskellPackages.haskell-language-server: 0.1.0.0 -> 0.2.0.0

generated by
pkgs/development/tools/haskell/haskell-language-server/update.sh

with some manual tweaks

twhitehead committed:
haskellPackages.scheduler: unmark as broken

twhitehead committed:
haskellPackages.massiv*: disable massiv test, unmark as broken

twhitehead committed:
haskellPackages.Color: unmark broken

ggreif committed:
ghc: add new version 8.8.4

basvandijk committed:
haskellPackages.splot: fix build

Also introduce the top-level `splot` package which only provides the
statically linked `splot` binary.

peti committed:
haskellPackages: bump default compiler to ghc-8.8.4

peti committed:
ghc: drop obsolete 8.4.x version

We generally keep only the last 3 major releases of GHC.

peti committed:
bluespec: mark package as broken because it depends on dropped ghc-8.4.4

Ping @flokli and @thoughtpolice.

peti committed:
sad: mark package as broken because if depends on dropped ghc-8.4.4

Ping @schmitthenner.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-10-gf433b07 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/dce52887aff29979bd9e5e01388ae97d1de02d5e.

domenkozar committed:
ghc 8.8.4, 8.10.1: disable profiled builds on aarch64 to avoid 2GB output limit

peti committed:
haskell-streaming-commons: update to version 2.2.2.1 to avoid test suite bug

maralorn committed:
haskellPackages.jsaddle-warp: Fix build

Disables a patch that is applied on upstream because of new version.

maralorn committed:
haskelllPackages.dependent-sum-aeson-orphans: apply patch to fix build

peti committed:
hackage2nix.yaml: update list of broken builds to avoid evaluation errors

Ping @maralorn, because neuron appears to be broken.

peti committed:
Merge remote-tracking branch 'origin/master' into haskell-updates.

maralorn committed:
haskellPackages.neuron: Fix build for 0.6.0 and deps

peti committed:
git-annex: update SHA256 hash for the new 8.20200720.1 version

maralorn commented:
@peti that was bad luck with the timing. I was working on a fix for neuron right then. Now I needed to rebase it on you marking it broken.^^

maralorn committed:
haskellPackages.haskell-language-server: 0.2.0.0 -> 0.2.2.0

maralorn committed:
Merge pull request #93762 from maralorn/fix-neuron

[haskell-updates] haskellPackages.neuron: Fix deps and build

maralorn committed:
Merge pull request #93768 from maralorn/hls

[haskell-updates] haskellPackages.haskell-language-server: 0.2.0.0 -> 0.2.2.0

maralorn commented:
Right now trying to fix the bluespec error.

maralorn committed:
bluespec: Fix build without ghc 844

ghc 8.4.4 was only required before the merge of

https://github.com/B-Lang-org/bsc/pull/42

which actually is already contained in the commit we are using.

peti committed:
Merge pull request #93772 from maralorn/fix-bluespec

[haskell-updates] bluespec: Fix build without ghc 844

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-11-g7fa9100 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/8af27e77a015d06c7a8fe49a430fd5334a93ebf7.

peti committed:
haskell-dhall: clean up and drop obsolete overrides

peti committed:
hackage2nix: update list of broken builds

Ping @cdepillabout because of termonad.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-11-g7fa9100-dirty from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/8af27e77a015d06c7a8fe49a430fd5334a93ebf7.

peti committed:
haskell-dhall: use latest version for ghc-8.10.x so that we can build with the latest repline library

peti committed:
all-cabal-hashes: update to Hackage at 2020-07-24T00:16:27Z

peti committed:
haskell-shower: jailbreak to fix the build with ghc-8.10.x

peti committed:
sad: comment out the broken build

I cannot mark this package broken in a way that doesn't break OfBorg.

peti commented:
Next week, we'll merge https://github.com/NixOS/nixpkgs/pull/94347.
